syntax = "proto2";
package proto;

message GeoGrid {
    message Location {
        required int64 id = 1;
        required string name = 2;
        required int64 province = 3;
        required int64 population = 4;
        message Coordinate {
            required double longitude = 1;
            required double latitude = 2;
        }
        required Coordinate coordinate = 5;
        message ContactCenter {
            required int64 id = 1;
            enum Type {
                K12School = 0;
                Community = 1;
                PrimaryCommunity = 2;
                SecondaryCommunity = 3;
                College = 4;
                Household = 5;
                Workplace = 6;
            }
            required Type type = 2;
            message ContactPool {
                required int64 id = 1;
                repeated int64 people = 2;
            }
            repeated ContactPool pools = 3;
        }
        repeated ContactCenter contactCenters = 6;
        message Commute {
            required int64 to = 1;
            required double proportion = 2;
        }
        repeated Commute commutes = 7;
    }
    message Person {
        required int64 id = 1;
        required int64 age = 2;
        required string gender = 3;
    }
    repeated Location locations = 1;
    repeated Person persons = 2;

}
